datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaBasedConnectorConfig.java

```diff
@@ -41,6 +41,7 @@
   public static final long DEFAULT_NON_GOOD_STATE_THRESHOLD_MILLIS = Duration.ofMinutes(10).toMillis();
   public static final long MIN_NON_GOOD_STATE_THRESHOLD_MILLIS = Duration.ofMinutes(1).toMillis();
 
```

Comment:
Please take reference example of `CONFIG_CONSUMER_FACTORY_CLASS` to create factory objects and validation of the class name.

It is better to use a factory as it allows setting different constructors in the future if required. 
If you decide to create a class object, even then, use the above config as reference.